src/v/datalake/translation/state_machine.cc

```diff
@@ -230,7 +230,7 @@ void translation_stm::update_highest_translated_offset(
 }
 
 bool stm_factory::is_applicable_for(const storage::ntp_config& config) const {
-    return model::is_user_topic(config.ntp());
+    return _iceberg_enabled && model::is_user_topic(config.ntp());
```

Comment:
This doesn't seem safe because the evaluation of iceberg_enabled is not synced across nodes. This means that different nodes can have this STM enabled and that we can end up with a split brain for the STM.

I'm kind of surprised this isn't something that can be optimized with the right local snapshot. Like, maybe when Iceberg isn't enabled, we can still add a snapshot that just indicates the point up to which Iceberg translation STM isn't enabled.

Reply:
How is this working right now with all the other Iceberg related services ? If a node is behind it will request to be restarted

Reply:
Posted in Slack, but this is probably fine at least to unblock upgrades, given how few folks are using Iceberg. I do think we should consider doing something different longer term though